[libSyntax][Cmake] Fix gyb invocation quoting issue on Windows

AndrewSB opened the pull request:
The lib/Syntax build system assumed a *nix quoting style for arguments
passed into gyb. That assumption breaks on Windows. This patch updates
the build system to Do The Right Thing™ when built on Windows.

harlanhaskins reviewed (approved):
Thanks!

harlanhaskins commented:
@swift-ci please test

harlanhaskins reviewed (changes requested):
Turns out this change makes the line directive become:

`"#line %(line)d "%(file)s""`, which no longer parses.

jrose-apple commented on lib/Syntax/CMakeLists.txt:
```diff
@@ -1,6 +1,7 @@
+
 set(SWIFT_GYB_FLAGS
-  --line-directive "'${line_directive}'")
+  --line-directive "\"${line_directive}\"")
```
I don't think this is correct; note that the actual body of `${line_directive}` contains quotes.

harlanhaskins replied:
@AndrewSB you might need to double-escape the quotes.

`”\\\”%(file)s\\\””`

AndrewSB replied:
parsing on parsing 😂

Thanks you two, I'll make the change now, and I can test it tomorrow during the day, when I'm back at my Windows machine

AndrewSB commented:
Made that change, can one of you ask CI to run again? But please don't merge until I verify that this continues to work on Windows.

harlanhaskins commented:
@swift-ci please smoke test

harlanhaskins commented:
Nice! Okay, now I wonder if CMake has some kind of “escape quotes” function we could use, instead of thjs

compnerd commented:
I don't think that there is any special quoting function. Maybe @llvm-beanz knows?

AndrewSB commented:
This currently breaks on windows, working on a fix now

AndrewSB committed:
gyb: On Windows, convert backslashes->slashes and quote filenames

gyb, when invoked through build system, ends up with arguments to
filepaths with backslashes (a Windows convention), instead of the
*nix-expected forward slashes to seperate directories. This change
converts those backslashes into forward slashes.

AndrewSB commented:
This worked on the latest clean build I tried, @harlanhaskins: can you re-review and run CI please?

harlanhaskins commented on cmake/modules/SwiftHandleGybSources.cmake:
```diff
@@ -57,24 +59,29 @@ function(handle_gyb_source_single dependency_out_var_name)
-      COMMENT "Generating ${basename} from ${GYB_SINGLE_SOURCE} ${GYB_SINGLE_COMMENT}"
-      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-      SOURCES "${GYB_SINGLE_SOURCE}"
-      IDEMPOTENT)
```
I'd really rather not reformat this file in this patch, just to make it easier to verify subtle differences didn't alter the CMake semantics.

AndrewSB replied:
I'll put it into a separate commit. Do you want me to make a new PR with it, or is keeping it in this one OK?

harlanhaskins replied:
It might(?) have broken the build as it is now? I think a new PR would be best, if you can.

AndrewSB replied:
I'll do it in a new one 😄 
It looks like the CI failed with an intermittent error, but I'm not 100% sure. Once I remove the formatting lets try again?

harlanhaskins commented:
Rest of the patch looks good 👍.

@swift-ci please clean test

AndrewSB committed:
libSyntax/cmake: Fix gyb invocation quoting issue on Windows

The lib/Syntax build system assumed a *nix quoting style for arguments
passed into gyb. That assumption breaks on Windows. This patch updates
the build system to Do The Right Thing™ when built on Windows.

AndrewSB commented:
@harlanhaskins moved the formatting changes into a different commit. I'll submit a PR after this is merged in so I don't have to deal with rebasing

Could you re-test?

harlanhaskins commented:
@swift-ci please clean test

harlanhaskins commented:
Ignore that spurious failure -- it's re-running the build.

AndrewSB commented:
Another infrastructure failure? It reported seconds after you asked and I don't see an obvious gyb issue in the logs

harlanhaskins commented:
@swift-ci please clean test
